compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -253,20 +253,28 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for DeeplyNormalizeForDiagnosticsFolder<'_,
     }
 
     fn fold_ty(&mut self, ty: Ty<'tcx>) -> Ty<'tcx> {
-        deeply_normalize_with_skipped_universes(
-            self.at,
-            ty,
-            vec![None; ty.outer_exclusive_binder().as_usize()],
-        )
-        .unwrap_or_else(|_: Vec<ScrubbedTraitError<'tcx>>| ty.super_fold_with(self))
+        self.at
+            .infcx
+            .commit_if_ok(|_| {
```

Comment:
```suggestion
        let infcx = self.at.infcx;
        infcx.commit_if_ok(|_| {
```

Reply:
or does this still not solve the ugly formatting due to the unwrap or else call?

Reply:
Doesn't change anything about the unwrap call.